### PR TITLE
IGVF-268-ensembl-gene-id

### PR DESCRIPTION
**Pull request by khzlin**

### CODE REVIEW

File: src/igvfd/schemas/changelogs/gene.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 ## Changelog for *`gene.json`*
 
+### Schema version 3
+
+* Remove `ncbi_entrez_status`.
+* Change `geneID` to use ENCSEMBL ID instead of NCBI Entrez ID.
```
Comment (gabdank) [inline src/igvfd/schemas/changelogs/gene.md:6]: typo: ENCSEMBL -> ENSEMBL

File: src/igvfd/schemas/gene.json [hunk 2]
```diff
@@ -140,7 +127,7 @@
                 "description": "A unique identifier from external resource (e.g. HGNC, MGI, FlyBase, WormBase, ENSEMBL, MIM, UniProtKB, Vega, miRBase).",
                 "comment": "Submit as database_name:id. See changelog for potential databases.",
                 "type": "string",
-                "pattern": "^((HGNC:\\d+)|(MGI:\\d+)|(ENSEMBL:ENS[A-Z]*G\\d+)|(MIM:\\d+)|(UniProtKB:[0-9A-Z]+)|(Vega:OTT[A-Z]+G\\d+)|(miRBase:MI\\d+)|(IMGT/GENE-DB:[0-9A-Za-z/\\(\\)\\-]+)|(RefSeq:[A-Z]{2}_[\\d.]+)|(GeneCards:[0-9a-zA-Z-]+))$"
+                "pattern": "^((HGNC:\\d+)|(MGI:\\d+)|(MIM:\\d+)|(UniProtKB:[0-9A-Z]+)|(Vega:OTT[A-Z]+G\\d+)|(miRBase:MI\\d+)|(IMGT/GENE-DB:[0-9A-Za-z/\\(\\)\\-]+)|(RefSeq:[A-Z]{2}_[\\d.]+)|(GeneCards:[0-9a-zA-Z-]+))$"
```
Comment (gabdank) [inline src/igvfd/schemas/gene.json:130]: please add to the regex the ENTREZ regex

File: src/igvfd/upgrade/gene.py [hunk 3]
```diff
@@ -7,3 +7,20 @@ def gene_1_2(value, system):
...
+    if 'ncbi_entrez_status' in value:
+        del value['ncbi_entrez_status']
+    if 'dbxrefs' in value:
+        for dbxref in value['dbxrefs']:
+            if dbxref.startswith('ENSEMBL:') == True:
+                dbxref = dbxref.removeprefix('ENSEMBL:')
+                value['geneid'] = dbxref
+            else:
+                no_ensembl.append(dbxref)
+    value.pop('dbxrefs')
```
Comment (gabdank) [inline src/igvfd/upgrade/gene.py:25]: I am not sure there is a need to "pop" if you are assigning the value in the next line

**Review by gabdank — CHANGES_REQUESTED** [review]

**Review by gabdank — APPROVED** [review]

**Comment by khzlin** [comment]
Closing ticket, Having trouble with making demos so I made a new branch at IGVF-268-ensembl-gene-id-b.

**Pull request closed without merging**